wgpu-hal/src/metal/adapter.rs

```diff
@@ -718,8 +718,12 @@ impl super::PrivateCapabilities {
             format_min_srgb_channels: if os_is_mac { 4 } else { 1 },
             format_b5: !os_is_mac,
             format_bc: os_is_mac,
-            format_eac_etc: !os_is_mac,
-            format_astc: Self::supports_any(device, ASTC_PIXEL_FORMAT_FEATURES),
+            format_eac_etc: !os_is_mac
+                // M1 in macOS supports EAC/ETC2
+                || device.supports_family(MTLGPUFamily::Apple7),
```

Comment:
we can only call `supports_family` if `family_check` is true

Reply:
Done.